### PR TITLE
added Movie model, Dao, Service

### PR DESCRIPTION
**Pull request by yutsevich**

### CODE REVIEW

File: src/main/java/com/dev/cinema/lib/Dao.java [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package com.dev.cinema.lib;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+
+@Retention(RetentionPolicy.RUNTIME)
```
Comment (andriylagun) [inline src/main/java/com/dev/cinema/lib/Dao.java:6]: Maybe better will use @Target(ElementType.TYPE) but not sure))

**Review by andriylagun — APPROVED** [review]
Looks nice!

File: src/main/java/com/dev/cinema/lib/Dao.java (hunk 1 above)
Comment (dmytro-hryhoruk) [reply src/main/java/com/dev/cinema/lib/Dao.java:6]: let's leave as it is

File: src/main/java/com/dev/cinema/dao/impl/MovieDaoImpl.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+            transaction = session.beginTransaction();
+            Long movieId = (Long) session.save(movie);
+            transaction.commit();
+            movie.setId(movieId);
+            return movie;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new RuntimeException("Can not insert movie entity " + movie.toString(), e);
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/dev/cinema/dao/impl/MovieDaoImpl.java]: let's use custom DataProcessingException in dao layer

File: src/main/java/com/dev/cinema/dao/impl/MovieDaoImpl.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+import java.util.List;
+
+@Dao
+public class MovieDaoImpl implements MovieDao {
+
+    @Override
+    public Movie add(Movie movie) {
+        Transaction transaction = null;
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/dev/cinema/dao/impl/MovieDaoImpl.java]: https://mate-academy.github.io/jv-program-fulltime/03_hibernate/checklist/hibernate_01.html

File: src/main/java/com/dev/cinema/dao/impl/MovieDaoImpl.java [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    @Override
+    public Movie add(Movie movie) {
+        Transaction transaction = null;
+        Session session = null;
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            Long movieId = (Long) session.save(movie);
+            transaction.commit();
+            movie.setId(movieId);
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/dev/cinema/dao/impl/MovieDaoImpl.java]: we don't need to do this manually, hibernate does it) test it and fix

File: src/main/java/com/dev/cinema/util/HibernateUtil.java [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+    private static SessionFactory sessionFactory = initSessionFactory();
+
+    private HibernateUtil() {
+    }
+
+    private static SessionFactory initSessionFactory() {
+        try {
+            return new Configuration().configure().buildSessionFactory();
+        } catch (Exception e) {
+            throw new DataProcessingException("Can not create Session Factory ", e);
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/dev/cinema/util/HibernateUtil.java]: let's use Runtime here , it's a different type of exception compare to Dao layer

File: src/main/java/com/dev/cinema/dao/impl/MovieDaoImpl.java [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+public class MovieDaoImpl implements MovieDao {
+
+    @Override
+    public Movie add(Movie movie) {
+        Transaction transaction = null;
+        Session session = null;
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            Long movieId = (Long) session.save(movie);
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/dev/cinema/dao/impl/MovieDaoImpl.java]: do you need to store the result in variable which you don't use?

**Review by dmytro-hryhoruk — APPROVED** [review]

**Review by dmytro-hryhoruk — APPROVED** [review]

**Pull request merged**